src/plugProjectNishimuraU/RedKabutoMgr.cpp

```diff
@@ -1,429 +1,71 @@
-	lwz      r30, 8(r1)
-	mtlr     r0
-	addi     r1, r1, 0x10
-	blr
-	*/
-}
+void Mgr::createObj(int count) { m_obj = new Obj[count]; }
 
 /*
  * --INFO--
  * Address:	802FFFE8
  * Size:	000010
  */
-void RedKabuto::Mgr::getEnemy(int)
-{
-	/*
-	mulli    r0, r4, 0x2f4
-	lwz      r3, 0x48(r3)
-	add      r3, r3, r0
-	blr
-	*/
-}
+EnemyBase* Mgr::getEnemy(int idx) { return &m_obj[idx]; }
 
 /*
  * --INFO--
  * Address:	802FFFF8
  * Size:	000080
  */
-void RedKabuto::Mgr::loadTexData()
+void Mgr::loadTexData()
 {
-	/*
-	stwu     r1, -0x40(r1)
-	mflr     r0
-	stw      r0, 0x44(r1)
-	li       r0, 0
-	stw      r31, 0x3c(r1)
-	mr       r31, r3
-	stw      r0, 0x44(r3)
-	addi     r3, r1, 8
-	lwz      r4,
-"cKabutoChangeTexName__Q34Game9RedKabuto26@unnamed@RedKabutoMgr_cpp@"@sda21(r13)
-	bl       __ct__Q212LoadResource3ArgFPCc
-	lwz      r3, gLoadResourceMgr@sda21(r13)
-	addi     r4, r1, 8
-	bl       load__Q212LoadResource3MgrFRQ212LoadResource3Arg
-	cmplwi   r3, 0
-	beq      lbl_8030003C
-	lwz      r0, 0x30(r3)
-	stw      r0, 0x44(r31)
-
-lbl_8030003C:
-	lwz      r0, 0x44(r31)
-	cmplwi   r0, 0
-	bne      lbl_80300064
-	lis      r3, lbl_8048D8D4@ha
-	lis      r5, lbl_8048D8E8@ha
-	addi     r3, r3, lbl_8048D8D4@l
-	li       r4, 0x52
-	addi     r5, r5, lbl_8048D8E8@l
-	crclr    6
-	bl       panic_f__12JUTExceptionFPCciPCce
+	m_changeTexture = nullptr;
 
-lbl_80300064:
-	lwz      r0, 0x44(r1)
-	lwz      r31, 0x3c(r1)
-	mtlr     r0
-	addi     r1, r1, 0x40
-	blr
-	*/
-}
-
-/*
- * --INFO--
- * Address:	80300078
- * Size:	0000C8
- */
-RedKabuto::Mgr::~Mgr()
-{
-	/*
-	stwu     r1, -0x10(r1)
-	mflr     r0
-	stw      r0, 0x14(r1)
-	stw      r31, 0xc(r1)
-	mr       r31, r4
-	stw      r30, 8(r1)
-	or.      r30, r3, r3
-	beq      lbl_80300124
-	lis      r3, __vt__Q34Game9RedKabuto3Mgr@ha
-	addi     r3, r3, __vt__Q34Game9RedKabuto3Mgr@l
-	stw      r3, 0(r30)
-	addi     r0, r3, 0x38
-	stw      r0, 4(r30)
-	beq      lbl_80300114
-	lis      r3, __vt__Q34Game6Kabuto3Mgr@ha
-	addi     r3, r3, __vt__Q34Game6Kabuto3Mgr@l
-	stw      r3, 0(r30)
-	addi     r0, r3, 0x38
-	stw      r0, 4(r30)
-	beq      lbl_80300114
-	lis      r3, __vt__Q24Game12EnemyMgrBase@ha
-	addi     r3, r3, __vt__Q24Game12EnemyMgrBase@l
-	stw      r3, 0(r30)
-	addi     r0, r3, 0x38
-	stw      r0, 4(r30)
-	beq      lbl_80300114
-	lis      r3, __vt__Q24Game13IEnemyMgrBase@ha
-	addic.   r0, r30, 4
-	addi     r3, r3, __vt__Q24Game13IEnemyMgrBase@l
-	stw      r3, 0(r30)
-	addi     r0, r3, 0x38
-	stw      r0, 4(r30)
-	beq      lbl_80300114
-	lis      r4, __vt__16GenericContainer@ha
-	addi     r3, r30, 4
-	addi     r0, r4, __vt__16GenericContainer@l
-	li       r4, 0
-	stw      r0, 4(r30)
-	bl       __dt__5CNodeFv
+	LoadResource::Arg loadArg(cKabutoChangeTexName);
+	LoadResource::Node* resource = gLoadResourceMgr->load(loadArg);
 
-lbl_80300114:
-	extsh.   r0, r31
-	ble      lbl_80300124
-	mr       r3, r30
-	bl       __dl__FPv
+	if (resource != nullptr) {
+		m_changeTexture = (ResTIMG*)resource->_30;
```

Comment:
the rest of the casts in this PR are `static_cast`. we should probably standardize these

Reply:
Thanks for picking that up! Have updated it to a static_cast instead in all three spots